### PR TITLE
✨Add Interfaces for ManualTask

### PR DESCRIPTION
**Pull request by alexanderkasten**
**Changes:**

1. Add ManualTask SystemEvents
2. Add ManualTask EventAggregator Settings

**Issues:**

Closes #process-engine/process_engine_runtime#91

PR: #81 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/runtime/messages/event_aggregator_settings.ts [hunk 1]
```diff
@@ -14,15 +16,23 @@ const routeParams = {
...
+  manualTaskId: ':manual_task_id',
   signalReference: ':signal_ref',
 };
 
-// tslint:disable:max-line-length
 const routePaths = {
   finishUserTask:
-    `/processengine/correlation/${routeParams.correlationId}/processinstance/${routeParams.processInstanceId}/usertask/${routeParams.flowNodeInstanceId}/finish`,
+    `/processengine/correlation/${routeParams.correlationId}/processinstance/
+    ${routeParams.processInstanceId}/usertask/${routeParams.flowNodeInstanceId}/finish`,
```
Comment (LeoTT) [inline src/runtime/messages/event_aggregator_settings.ts]: Dont use newlines within template strings. Template strings will contain the break and spaces.
```suggestion
    `/processengine/correlation/${routeParams.correlationId}/processinstance/` +
    `${routeParams.processInstanceId}/usertask/${routeParams.flowNodeInstanceId}/finish`,
```

see https://github.com/5minds/javascript-guidelines/tree/develop/styleguide#strings--line-length

File: src/runtime/messages/event_aggregator_settings.ts [hunk 2]
```diff
@@ -14,15 +16,23 @@ const routeParams = {
...
   userTaskFinished:
-    `/processengine/correlation/${routeParams.correlationId}/processinstance/${routeParams.processInstanceId}/usertask/${routeParams.flowNodeInstanceId}/finished`,
+    `/processengine/correlation/${routeParams.correlationId}/processinstance/
+    ${routeParams.processInstanceId}/usertask/${routeParams.flowNodeInstanceId}/finished`,
```
Comment (LeoTT) [inline src/runtime/messages/event_aggregator_settings.ts]: see above

File: src/runtime/messages/event_aggregator_settings.ts [hunk 3]
```diff
@@ -14,15 +16,23 @@ const routeParams = {
...
+  finishManualTask:
+  `/processengine/correlation/${routeParams.correlationId}/processinstance/
+  ${routeParams.processInstanceId}/manualtask/${routeParams.flowNodeInstanceId}/finish`,
```
Comment (LeoTT) [inline src/runtime/messages/event_aggregator_settings.ts]: see above

File: src/runtime/messages/event_aggregator_settings.ts [hunk 4]
```diff
@@ -14,15 +16,23 @@ const routeParams = {
...
+  manualTaskFinished:
+    `/processengine/correlation/${routeParams.correlationId}/processinstance/
+    ${routeParams.processInstanceId}/manualtask/${routeParams.flowNodeInstanceId}/finished`,
```
Comment (LeoTT) [inline src/runtime/messages/event_aggregator_settings.ts]: see above

File: src/runtime/messages/system_events/manual_task_finished.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import {BaseSystemEventMessage} from './base_system_event_message';
+
+/**
+ * The message sent when a manual task has been finished.
```
Comment (robpal94) [inline src/runtime/messages/system_events/manual_task_finished.ts]:
```suggestion
 * Represents the message that is sent, when a ManualTask has been finished.
```

File: src/runtime/messages/system_events/manual_task_reached.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import {BaseSystemEventMessage} from './base_system_event_message';
+
+/**
+ * The message sent when a manual task has been finished.
```
Comment (robpal94) [inline src/runtime/messages/system_events/manual_task_reached.ts]:
```suggestion
 * Represents the message which is send, when a ManualTask has been finished.
```

**Pull request merged**